### DISCUSSION TITLE
XPM2 support

### DISCUSSION THREAD

**Discussion by Timmmm** [ImageMagick/ImageMagick#6718] [Contribute]
It seems ImageMagick supports XPM3 but not XPM2 (see [wikipedia](https://en.wikipedia.org/wiki/X_PixMap) for the difference). Actually that's almost all the information there is about XPM2 - the only piece of software I've found that supports it is `sxpm` which is part of [libXpm](https://github.com/freedesktop/libXpm) itself.

So why on earth would you want this? Well it's very convenient for my current use-case which is embedding an image with only a few colours (6) into a JSON object. It's ASCII-only so it happily goes in a string and I don't need to base64 encode it, plus it is a reasonably efficient way to store 6-colour images (1 byte per pixel), and it is very very simple.

I can't find any other image formats that have those properties.

**Comment by urban-warrior** [comment]
We are not inclined to add XPM2 support to ImageMagick, at least not anytime soon.  However, since ImageMagick is open-source, any user can contribute support with a Pull request.
